Fix stereo camera resolution

h-wata opened the pull request:
* Adapted to the resolution of the OAK-D device. 
* fix #18 .

h-wata committed:
Fix stereo camera resolution

saching13 reviewed (changes requested):
Thanks a lot for the PR. And sorry for creating the bug in the first place. If possible can you make the resolution selectable from an argument with 720 as default?

saching13 commented:
and also use the method [here](https://github.com/luxonis/depthai-ros-examples/blob/e9f2fd604f0cee291e30152be57fc7283f6f12c8/depthai_examples/ros1_src/stereo_nodelet.cpp#L83) to get the calibInfo from the device instead of the default files.

h-wata committed:
fix: Selecatable stereo camera resolution.

h-wata commented:
Thank you for your review. 
In response to your comment, I have modified the resolution so that you can select it from rosparam.

> Thanks a lot for the PR. And sorry for creating the bug in the first place. If possible can you make the resolution selectable from an argument with 720 as default?

But I don't have a ROS2 environment, so I can't verify whether the build will pass or not, so I need to test it in ROS2 environment.

I will work on the second comment.

> and also use the method [here](https://github.com/luxonis/depthai-ros-examples/blob/e9f2fd604f0cee291e30152be57fc7283f6f12c8/depthai_examples/ros1_src/stereo_nodelet.cpp#L83) to get the calibInfo from the device instead of the default files.

saching13 commented:
> Thank you for your review. In response to your comment, I have modified the resolution so that you can select it from rosparam.
> 
> > Thanks a lot for the PR. And sorry for creating the bug in the first place. If possible can you make the resolution selectable from an argument with 720 as default?
> 
> But I don't have a ROS2 environment, so I can't verify whether the build will pass or not, so I need to test it in ROS2 environment.
> 
> I will work on the second comment.
> 
> > and also use the method [here](https://github.com/luxonis/depthai-ros-examples/blob/e9f2fd604f0cee291e30152be57fc7283f6f12c8/depthai_examples/ros1_src/stereo_nodelet.cpp#L83) to get the calibInfo from the device instead of the default files.

No worries on ROS2 env. LMK after the changes. I will test the ROS2 and get back to you.

h-wata committed:
Read camera config from device

h-wata commented:
It's ready, please review it at your convenience.

themarpe commented on depthai_examples/ros1_src/yolov4_spatial_publisher.cpp:
```diff
@@ -1,6 +1,7 @@
 
 #include "ros/ros.h"
 
+#include <depthai-shared/properties/MonoCameraProperties.hpp>
```
Include  
```
#include <depthai/pipeline/node/MonoCamera.hpp>
```

And use `dai::node::MonoCamera::Properties::SensorResolution`, etc... instead

h-wata replied:
Fixed in b9b6b81.

saching13 commented on depthai_examples/ros1_src/yolov4_spatial_publisher.cpp:
```diff
@@ -125,14 +140,20 @@ int main(int argc, char** argv){
 
-    dai::rosBridge::ImageConverter rgbConverter(deviceName + "_rgb_camera_optical_frame", false);
-    dai::rosBridge::BridgePublisher<sensor_msgs::Image, dai::ImgFrame> rgbPublish(device.getOutputQueue("preview", 30, false),
+    dai::rosBridge::BridgePublisher<sensor_msgs::Image, dai::ImgFrame> rgbPublish(colorQueue,
```
`auto rgbCameraInfo = rgbtConverter->calibrationToCameraInfo(calibrationHandler, dai::CameraBoardSocket::RGB, width, height)`

You can do the same for RGB camera too instead of using color.yaml

h-wata replied:
Thanks.
I apologize for my lack of understanding.
Should the height and width match the preview(416,416) or the 1080p(1920, 1080)?

h-wata replied:
Fixed in b9b6b81. 
CameraInfo match (416, 416). If I misunderstood, please comment.

saching13 commented on depthai_examples/ros1_src/yolov4_spatial_publisher.cpp:
```diff
@@ -51,9 +56,17 @@ dai::Pipeline createPipeline(bool syncNN, bool subpixel, std::string nnPath, int
+        monoResolution = dai::MonoCameraProperties::SensorResolution::THE_720_P; 
+    }else if(resolution == "400p" ){
+        monoResolution = dai::MonoCameraProperties::SensorResolution::THE_400_P; 
+    }else{
+        monoResolution = dai::MonoCameraProperties::SensorResolution::THE_800_P; 
+    }
```
can you validate if `800p` is passed on the string too ? and throw an error on else if it has an invalid string. 
And there is 480p res too (Which is for OAK-D-LITE) we can add that too here.

h-wata replied:
Fixed in b9b6b81.

saching13 commented on depthai_examples/ros2_src/yolov4_spatial_publisher.cpp:
```diff
@@ -51,9 +52,16 @@ dai::Pipeline createPipeline(bool syncNN, bool subpixel, std::string nnPath, int
+        monoResolution = dai::MonoCameraProperties::SensorResolution::THE_720_P; 
+    }else if(resolution == "400p" ){
+        monoResolution = dai::MonoCameraProperties::SensorResolution::THE_400_P; 
+    }else{
+        monoResolution = dai::MonoCameraProperties::SensorResolution::THE_800_P; 
+    }
```
Same as in ros1

h-wata replied:
Fixed in b9b6b81.

saching13 commented on depthai_examples/ros2_src/yolov4_spatial_publisher.cpp:
```diff
@@ -127,15 +138,20 @@ int main(int argc, char** argv){
     //TODO(sachin): Add option to use CameraInfo from EEPROM
     dai::rosBridge::ImageConverter rgbConverter(deviceName + "_rgb_camera_optical_frame", false);
-    dai::rosBridge::BridgePublisher<sensor_msgs::msg::Image, dai::ImgFrame> rgbPublish(device.getOutputQueue("preview", 30, false),
+    dai::rosBridge::BridgePublisher<sensor_msgs::msg::Image, dai::ImgFrame> rgbPublish(colorQueue,
```
Same as in ros1

h-wata replied:
Fixed in b9b6b81.

h-wata committed:
fix for review

saching13 commented on depthai_examples/ros1_src/yolov4_spatial_publisher.cpp:
```diff
@@ -28,9 +29,13 @@ const std::vector<std::string> label_map = {"person",         "bicycle",    "car
              "toaster",        "sink",       "refrigerator",  "book",          "clock",       "vase",          "scissors",
              "teddy bear",     "hair drier", "toothbrush"};
+std::unique_ptr<dai::rosBridge::BridgePublisher<sensor_msgs::Image, dai::ImgFrame>> rgbPublish, depthPublish;
+std::unique_ptr<dai::rosBridge::ImageConverter> rgbConverter, rightConverter;
```
Any specific reason why unique ptr is created here ?

h-wata replied:
The following parts have been followed. 
https://github.com/luxonis/depthai-ros-examples/blob/e9f2fd604f0cee291e30152be57fc7283f6f12c8/depthai_examples/ros1_src/stereo_nodelet.cpp#L21-L23
There is no particular reason to do otherwise. 
If it is unnecessary, we will correct it.

h-wata replied:
I checked it works with OAK-D and OAK-D-LITE.
The unique_ptr was being used unnecessarily, so I changed it back.

h-wata committed:
fix: test oak-d and oak-d-lite

I checked it works with OAK-D and OAK-D-LITE.
The unique_ptr was being used unnecessarily, so I changed it back.
